From 0dc56a4ea2192c6724c23a730476aa9b2af5657b Mon Sep 17 00:00:00 2001
From: Ludwig Krispenz <lkrispen@redhat.com>
Date: Fri, 14 Oct 2016 13:50:18 +0200
Subject: [PATCH 410/410] Ticket 49009 - args debug logging must be more
 restrictive

Bug Description: turning on args debugging logs all attribute value, including #unhashed#

Fix Description: filter unhashed attrs

https://fedorahosted.org/389/ticket/49009

Reviewed by: ?

(cherry picked from commit ca0d132acbd26c16501540411b198cd2a3b71252)
---
 ldap/servers/slapd/entry.c    | 1 +
 ldap/servers/slapd/entrywsi.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/ldap/servers/slapd/entry.c b/ldap/servers/slapd/entry.c
index f3d9c86..09671a4 100644
--- a/ldap/servers/slapd/entry.c
+++ b/ldap/servers/slapd/entry.c
@@ -3332,6 +3332,7 @@ entry_apply_mod( Slapi_Entry *e, const LDAPMod *mod )
 	  if((strcasecmp(mod->mod_type,"objectclass") == 0)  
               && (strncasecmp((const char *)mod->mod_bvalues[i]->bv_val,"ldapsubentry",mod->mod_bvalues[i]->bv_len) == 0)) 
 	    sawsubentry=PR_TRUE;
+	  if (0==strcasecmp(PSEUDO_ATTR_UNHASHEDUSERPASSWORD,mod->mod_type)) continue;
 	  LDAPDebug( LDAP_DEBUG_ARGS, "   %s: %s\n", mod->mod_type, mod->mod_bvalues[i]->bv_val, 0 );
 	}
 
diff --git a/ldap/servers/slapd/entrywsi.c b/ldap/servers/slapd/entrywsi.c
index 76286a1..3edca33 100644
--- a/ldap/servers/slapd/entrywsi.c
+++ b/ldap/servers/slapd/entrywsi.c
@@ -775,6 +775,7 @@ entry_apply_mod_wsi(Slapi_Entry *e, const LDAPMod *mod, const CSN *csn, int urp)
 		for ( i = 0;
 		      mod->mod_bvalues != NULL && mod->mod_bvalues[i] != NULL;
 		      i++ ) {
+			if (0==strcasecmp(PSEUDO_ATTR_UNHASHEDUSERPASSWORD,mod->mod_type)) continue;
 			LDAPDebug( LDAP_DEBUG_ARGS, "   %s: %s\n",
 			           mod->mod_type, mod->mod_bvalues[i]->bv_val, 0 );
 		}
-- 
2.4.11

